Client/TabManagement/TabMigrationUtility.swift

```diff
@@ -0,0 +1,69 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import Shared
+import TabDataStore
+
+protocol TabMigrationUtility {
+    var shouldRunMigration: Bool { get }
+    func runMigration(savedTabs: [LegacySavedTab]) async
+}
+
+class DefaultTabMigrationUtility: TabMigrationUtility {
+    private let migrationKey = PrefsKeys.TabMigrationKey
+    private var profile: Profile
+    private var tabDataStore: TabDataStore
+
+    init(profile: Profile = AppContainer.shared.resolve(),
```

Comment:
Since we only use the prefs from the Profile object in this Utility, we could only depends from the `Prefs` object rather than the whole `Profile` ?